Fixes for types serialization and deserialization

XVilka opened the pull request:
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes three bugs:
1. Atomic type formats were saved as `type.bool=b bool bool` while it should be `type.bool=b`
2. Bit-specific types (`types-16.sdb` for example) were loaded only if `arch` was defined as well, we fix the order now.
3. Functions were loaded before types, which produced an error if function local variable used type defined in the `/core/analysis/types` project section.

**Test plan**

CI is green

**Closing issues**

Fixes https://github.com/rizinorg/rizin/issues/1560

XVilka committed:
Fix atomic type format serialization

XVilka committed:
Load bit-specific types before arch-specific

XVilka committed:
Load types, callables, classes before functions and variables

wargio commented:
is there a test for this?

ret2libc reviewed (changes requested):
It seems I can still reproduce the problem in the referenced issue.

ret2libc commented:
[ls.rzdb.txt](https://github.com/rizinorg/rizin/files/7115838/ls.rzdb.txt)

XVilka committed:
Bump Project version to 5, add `unknown_t`

XVilka commented:
@ret2libc it was because projects created before used `unknown_t` in the function, variable types, but not added in the `/core/analysis/types` section of the project. I bumped the version of project to 5 and added a migration step that adds the corresponding types. Added also the unit test for checking if it was added. If you have more ideas how to test this better - let me know.

ret2libc commented:
> @ret2libc it was because projects created before used `unknown_t` in the function, variable types, but not added in the `/core/analysis/types` section of the project. I bumped the version of project to 5 and added a migration step that adds the corresponding types. Added also the unit test for checking if it was added. If you have more ideas how to test this better - let me know.

Ok I think it kinda works now. The first time you load and save it it still gives a lot of `Cannot find base type "wchar_t"` errors, but after that first time it doesn't print errors anymore. It's ok, I think. Maybe wait for @thestr4ng3r to check the migration as well.
